aea/helpers/pypi_checker.py

```diff
@@ -0,0 +1,297 @@
+        return not one_of_them_is_a_strict_comparison
+    else:
+        return True
+
+
+class PyPIChecker:
+    """
+    Find out if PyPI version specifiers are unsatisfiable.
+
+    Its usage is incremental: you keep an instance of the
+    checker and you populate with new constraints.
+    """
+
+    def __init__(self):
+        """Initialize the checker."""
+        # mapping from package name to specifier set.
+        self._pypi_dependencies = {}  # type: Dict[str, SpecifierSet]
+
+    def add(self, package_name: str, specifier_set: SpecifierSet):
+        """
+        Add a specifier set for a package in the checker.
+
+        :param package_name: the package name
+        :param specifier_set: the set of specifiers.
+        :return: None
+        :raises ValueError: if the set of specifier plus the one already present
+            is surely unsatisfiable.
+        """
+        specifier = self._merge_with_existing_specifier_set_if_any(
+            package_name, specifier_set
+        )
+
+        try:
+            result = _is_satisfiable(specifier)
+        except Exception:
+            # we ignore possible errors. There might be some corner cases that make
+            # our checks to raise exception, e.g. requirements like git+https://...
+            result = True
+
+        if result is False:
+            raise ValueError(
+                "Specifier set {} for package {} not satisfiable.".format(
+                    specifier_set, package_name
+                )
+            )
+        self._pypi_dependencies[package_name] = specifier_set
+
+    def _add_or_merge_with_existing_specifier_set(
+        self, package_name: str, specifier_set: SpecifierSet
+    ) -> None:
+        """
+        Add a new package/specifier set pair, or, if one already exists, merge
+        the new specifier with the old one.
+
+        :param package_name: the package name
+        :param specifier_set: the specifier set
+        :return: None
+        """
+        if package_name in self._pypi_dependencies:
+            previous_spec = self._pypi_dependencies[package_name]
+            new_spec = operator.and_(previous_spec, specifier_set)
+            self._pypi_dependencies[package_name] = new_spec
+        else:
+            self._pypi_dependencies[package_name] = specifier_set
+
+    def _merge_with_existing_specifier_set_if_any(
+        self, package_name: str, new_specifier_set: SpecifierSet
+    ) -> SpecifierSet:
+        """
+        If a package name is already registered, take the associated specifier set
+        and merge with the new one.
+        Otherwise, return the new one.
+
+        :param package_name: the package name
+        :param new_specifier_set: the new specifier set.
+        :return: the merged specifier set, or only the new one.
+        """
+        if package_name in self._pypi_dependencies:
+            old_set = self._pypi_dependencies[package_name]
+            return and_(old_set, new_specifier_set)
```

Comment:
For consistency with the call from `_add_or_merge_with_existing_specifier_set`:
```suggestion
            return operator.and_(old_set, new_specifier_set)
```